x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/esql/esql.ts

```diff
@@ -131,7 +132,11 @@ export const esqlExecutor = async ({
           excludedDocumentIds: excludedDocuments.map(({ id }) => id),
           ruleExecutionTimeout,
         });
-        const esqlQueryString = { drop_null_columns: true };
+
+        const esqlQueryString = {
+          drop_null_columns: true,
+          allow_partial_results: !isRuleAggregating,
```

Comment:
Can we add a comment here just stating `allow_partial_results` defaults to true? I like to have it documented in-line.

Ref: https://github.com/elastic/elasticsearch/issues/122802